misk-redis/src/main/kotlin/misk/redis/RedisConfig.kt

```diff
@@ -2,8 +2,18 @@ package misk.redis
 
 import misk.config.Config
 
-data class RedisConfig(
-  val host_name: String,
-  val port: Int,
-  val auth_password: String
-) : Config
+class RedisConfig: java.util.LinkedHashMap<String, RedisReplicationGroupConfig>, Config {
+  constructor(): super()
+  constructor(m: Map<String, RedisReplicationGroupConfig>): super(m)
+}
+
+data class RedisReplicationGroupConfig(
+  val writer_endpoint: RedisNodeConfig,
+  val reader_endpoint: RedisNodeConfig,
+  val redis_auth_password: String
```

Comment:
I wonder if this should be a `Secret`. while it's not strictly necessary now because it's a single value, I would assume in the future `Secret` would allow us to redact the value instead of fuzzy matching on fields named `password`

Reply:
I think a `Secret` makes sense here, I'll keep it in mind